### ISSUE TITLE
Always asking for a password

### ISSUE THREAD

**Issue by Andy944** [fox-it/linux-luks-tpm-boot#9]
At the step of "you should not be asked for a password during boottime" while I am always asking for a password although I did not change anything in a boottime.. 
Not sure, where I made a mistake.

Could you please help me with this issue?

Thanks a lot!

**Comment by morbitzer** [comment]
Which passphrase do you need for booting? The passphrase for your encrypted partition, or a passphrase for your NVRAM area or your SRK? 
If you need your disk-encryption passphrase, I would guess that you forgot to edit `/etc/crypttab`.

**Comment by Andy944** [comment]
Yes I need my disk-encryption passphrase, always after rebooting..
my /etc/crypttab looks like this
sda2_crypt UUID=ca709269-1e3e-4e9e-9e08-7248f0e6c5a6 none luks,keyscript=/sbin/getsecret.sh

**Comment by morbitzer** [comment]
Ok, this probably means that the scripts are not working as they should. Are all scripts executable and in the correct location? Did you update your initramfs? 
You could also try appending `break=init` to your kernel command line so that you will get dropped into busybox. There, you can check if the `getsecret.sh` is correctly installed and gives you the passphrase.

**Comment by Andy944** [comment]
Yes, I followed your instruction, and everything should be in correct location.... so I checked in my busybox `getsecret.sh`
`(initramfs) /sbin/getsecret.sh`
`/sbin/getsecret.sh: line 9: head: not found`
`Tspi_NV_ReadValue failed: 0x00000008 - layer=tpm, code=0008 (8), The TPM target command has been disabled`
and this is what I got..

**Comment by morbitzer** [comment]
This seems to be the same issue as described in issue #7 . Could you try to follow the instructions of @JorritVrieze to see if this works for you? 

[https://github.com/fox-it/linux-luks-tpm-boot/issues/7#issuecomment-309708689](url)

**Comment by Andy944** [comment]
I tried it, (changed head -c into dd b=$SIZE count=1) and now `getsecret.sh` in busybox is working, but I can not properly see its output, because hexdump is not available command in busybox.. but still is keeps asking me for a passphrase, but I noticed, that everytime I encrypt the device, its UUID is changed after one reboot.

**Comment by Andy944** [comment]
After reboot, the UUID of encrypted device is changed (because of encrypting I suppose?), so in first reboot, it do not ask me for a passphrase, because there is obviously wrong UUID in `/etc/crypttab`, after that I rewrited old UUID with the UUID of encrypted device, and it keeps asking me for a passphrase.

**Comment by morbitzer** [comment]
The UUID should stay the same, also after reboot. Are you sure you are using the UUID of your encrypted home partition? You can check with `sudo blkid -o list`. You should be using the UUID of the partition that states `crypto_LUKS` for as `fs_type`. 

Alternatively to using UUIDs, you could try to use the devicename instead in `/etc/crypttab`, such as `/dev/sda5`

**Comment by Andy944** [comment]
Ok, I used devicename instead of UUID, and still is asking me for a passphrase :(
In busybox, `getsecred.sh` is giving me passphrase (I just see some symbols, because I can not use hexdump in kernel command)
